ufo2fontir/src/source.rs

```diff
@@ -1733,4 +1800,186 @@ mod tests {
+            outer
+        };
+
+        // When
+        let postscript_names =
+            postscript_names(&lib_plist).expect("postscript names should be parsed");
+
+        // Then
+        assert_eq!(
+            postscript_names,
+            HashMap::from_iter([
+                (GlyphName::from("foo"), GlyphName::from("qux")),
+                (GlyphName::from("lorem"), GlyphName::from("ipsum")),
+            ]),
+        );
+    }
+
+    #[test]
+    fn postscript_names_drops_non_strings() {
+        // Given
+        let lib_plist = {
+            let mut outer = plist::Dictionary::new();
+            let mut inner = plist::Dictionary::new();
+            inner.extend([
+                (String::from("foo"), String::from("bar").into()),
+                (String::from("baz"), 12f32.into()),
+                (String::from("lorem"), true.into()),
+            ]);
+            outer.insert(String::from("public.postscriptNames"), inner.into());
+            outer
+        };
+
+        // When
+        let postscript_names =
+            postscript_names(&lib_plist).expect("postscript names should be parsed");
+
+        // Then
+        assert_eq!(
+            postscript_names,
+            HashMap::from_iter([(GlyphName::from("foo"), GlyphName::from("bar"))]),
+        );
+    }
+
+    #[test]
+    fn postscript_names_allows_duplicate_values() {
+        // Given
+        let lib_plist = {
+            let mut outer = plist::Dictionary::new();
+            let mut inner = plist::Dictionary::new();
+            inner.extend([
+                (String::from("foo"), String::from("bar").into()),
+                (String::from("baz"), String::from("bar").into()),
+                (String::from("lorem"), String::from("bar").into()),
+            ]);
+            outer.insert(String::from("public.postscriptNames"), inner.into());
+            outer
+        };
+
+        // When
+        let postscript_names =
+            postscript_names(&lib_plist).expect("postscript names should be parsed");
+
+        // Then
+        assert_eq!(
+            postscript_names,
+            HashMap::from_iter([
+                (GlyphName::from("foo"), GlyphName::from("bar")),
+                (GlyphName::from("baz"), GlyphName::from("bar")),
+                (GlyphName::from("lorem"), GlyphName::from("bar")),
+            ]),
+        );
+    }
+
+    #[test]
+    fn static_metadata_loads_postscript_names() {
+        let (_, context) = build_static_metadata("designspace_from_glyphs/WghtVar.designspace");
+        let static_metadata = context.static_metadata.get();
+
+        assert_eq!(
+            static_metadata.postscript_names,
+            PostscriptNames::from_iter([(
+                GlyphName::from("manual-component"),
+                GlyphName::from("manualcomponent")
+            )]),
+        );
+    }
+
+    #[test]
+    fn static_metadata_disable_postscript_names() {
+        // Copied from build_static_metadata
```

Comment:
Can we have a shared function to reduce the noise and make it easier to see what's unique to this test instead of `// This is the line that's different from build_static_metadata`